src/main/osd/osd_elements.c

```diff
@@ -922,6 +922,18 @@ static void osdElementGpsSpeed(osdElementParms_t *element)
 {
     tfp_sprintf(element->buff, "%c%3d%c", SYM_SPEED, osdGetSpeedToSelectedUnit(gpsConfig()->gps_use_3d_speed ? gpsSol.speed3d : gpsSol.groundSpeed), osdGetSpeedToSelectedUnitSymbol());
 }
+
+static void osdElementEfficiency(osdElementParms_t *element)
+{
+    const int32_t speed = osdGetSpeedToSelectedUnit(gpsSol.groundSpeed);
+    if (sensors(SENSOR_GPS) && ARMING_FLAG(ARMED) && STATE(GPS_FIX) && speed > 0) {
```

Comment:
This should be rearranged to only calculate the speed inside this condition. Otherwise you're forcing the calculations and printf in `osdGetSpeedToSelectedUnit()` to run all the time and for cases where the user doesn't even have GPS set up.

Reply:
Also I don't think that a 0 speed should be a condition for the element to disappear. In that case the "efficiency" is 0. But you'll need to check for this to avoid a division by 0.